### ISSUE TITLE
Add 'scrollOverflow:false' while responsive.

### ISSUE THREAD

**Issue by nikkop** [alvarotrigo/fullPage.js#958]
I would like to use fullPage.js and it's autoScrolling feature on my site but there's one thing making me not using it.

When using 'responsive: value' the 'autoScrolling' feature is being turned off. The problem is that the 'scrollOverflow' feature remains unchanged from original. 

Would it be possible to turn off 'scrollOverflow' when 'responsive: value' is triggered? The reason I would like to do that is because I don't want to scroll within the '.section' in responsive mode, I just want it to act normal just like as if 'scrollOverflow' was turned off.

So basically;
How do I turn off 'scrollOverflow' when 'responsive' is triggered? It's very much needed for my design and I think it could be helpful for other too.

I would like fullPage.js with the features 'autoScrolling' and 'scrollOverflow' to be enabed above the responsive breakpoint. But when 'responsive' triggers I want 'autoScrolling' and 'scrollOverflow' turned off, both of them.

Thanks in advance

**Comment by alvarotrigo** [comment]
In any case, this issue is directly related with [this other one](https://github.com/alvarotrigo/fullPage.js/issues/553).

And not easy to solve. slimScroll.js is not perfect and fullPage.js might stop working it in a near future. Maybe changing to [iScroll.js](http://cubiq.org/iscroll-5), although this one is more heavy.

**Comment by spindiesel** [comment]
Hi Alvaro,

You've mentioned in other threads that if we add the following code, slimscroll won't be needed:

.fp-section{
     height: auto !important;
}

I've tried this multiple ways, however slim scroll still fires, and the sections still scroll. I cannot get the sections to extend beyond the height of the browser. Here is my js code:

$('#fullpage').fullpage({
	
resize: false,
scrollOverflow: true,
	anchors: ['home', 'about', 'work', 'fill1', 'fill2', 'contact'],
	menu: '#section_links',
	responsive: 1024 

});

Here is my css:

@media only screen and (max-width: 1024px) {

.fp-section{
    height: auto !important;
}

}

Is there anything else I need to be doing? My css file css is being included after the jquery.fullPage.css file.

**Comment by alvarotrigo** [comment]
@spindiesel as I said in my previous post, it is not easy to solve.

**Comment by richgcook** [comment]
@alvarotrigo @spindiesel Any news on this?

**Comment by alvarotrigo** [comment]
@richgcook yeah. About to be solved [thanks to @jnettik](https://github.com/alvarotrigo/fullPage.js/pull/1037). Probably merged in the minified slimscroll for the next released.

**Comment by richgcook** [comment]
@alvarotrigo Any way to get this now?

**Comment by alvarotrigo** [comment]
@richgcook if you really need it now, [contact me](http://alvarotrigo.com/#contact-page).

**Comment by richgcook** [comment]
@alvarotrigo I'm the guy who emailed you... thanks.

**Comment by welyanto** [comment]
i have the same issue for deleting or removing the scroll on my responsive before, after do some research this can works on me, maybe this can help for you to guys.
```javascript
if (document.documentElement.clientWidth > 1023) {
	$('#fullpage').fullpage({
		// we need to enable scrolling on desktop version
		scrollOverflow: true
		// the rest setting can goes below here
	});
}
if (document.documentElement.clientWidth < 1024) {
	$('#fullpage').fullpage({
		// we need to disable scrolling not on desktop version
		scrollOverflow: false
		// the rest setting can goes below here
	});
}
```

**Comment by alvarotrigo** [comment]
As the problem related with this issue was solved, I'll open it again now that it is possible to find a solution for it.

**Comment by reddo** [comment]
Is there a viable solution for this as of now? I'd like to disable scrollOverflow while window is smaller than responsiveWidth. By the way, I'm using the dev version together with iScroll.

**Comment by alvarotrigo** [comment]
@reddo nop. That's why the topic still open.

**Comment by reddo** [comment]
I had an idea: 
When you check if scrolloverflow needs to be added, also check if the window width is larger than or equal to responsiveWidth and only add it if true. I've found 2 instances of this check, on lines 407 and 579.

In my case, I also had to remove the inline style applied to my sections so the content doesn't overflow.

**Comment by reddo** [comment]
Sorry to bother you again, do you have any ideas on the last part of my last comment?

**Comment by alvarotrigo** [comment]
Not really.

**Comment by daverdalas** [comment]
Workaround that I use:
```javascript
var windowWidth = $(window).width();
    $('.fullpage').fullpage({
    responsiveWidth: 992,
    scrollOverflow: windowWidth > this.responsiveWidth
 });

**Comment by alvarotrigo** [comment]
Fixed in 2.8.0.

**Comment by syampolskiy** [comment]
it still does not work, maybe I have incorrect options for plugin?
`$('#block_for_scroll').fullpage({
		menu: ".nav_list",
		anchors: ['theplatform', 'yourbenefits', 'testimonies','contactus'],
		scrollOverflow: true,
		paddingTop: '130px',
		responsiveWidth: 991,
		scrollOverflowOptions: {
			scrollbars: true,
	        mouseWheel: true,
	        hideScrollbars: false,
	        fadeScrollbars: false,
	        disableMouse: true,
	        interactiveScrollbars: true
		}
	});`

**Comment by alvarotrigo** [comment]
> it still does not work,

You are right. The problem behind the impediment for this this was solved, but the feature was not yet implemented as you can see in this [reproduction online](http://jsfiddle.net/97tbk/1298/).

I'll keep it open until I add the feature in the next release.

**Comment by faliqaiman** [comment]
<pre>
$(document).ready(function() {
    $('#fullpage').fullpage({
		sectionsColor : ['#111', '#333', '#555', '#777'],
        responsiveWidth: 480,
        responsiveHeight: 480
	});
});
</pre>

Why when I adjust the width and height to 480px, it's not reflected on the mobile? But when I set it to ` responsiveWidth: 1000,` it's goes well. My plan is to have both height and width so when user using the mobile in portrait or landscape mode, fullpage function won't runs. Pls advise. Thanks.

**Comment by faliqaiman** [comment]
I try with `fullpage.js` ver2.7.5 and `responsiveWidth` function is okay. This the latest version is not support the `responsiveWidth`?

**Comment by Akkuolina** [comment]
I solved so simply, if you can help:

**// before running the fullpage**
var scrolloverflowed;
if ($('#fullpage').width() < 1024) { // check the width (or window.width() or other method)
 scrolloverflowed = false;
 } else {
 scrolloverflowed = true;
}

**// fullpage init**
$('#fullpage').fullpage({
...
...
scrollOverflow: scrolloverflowed,
...
...
});

_Akkuolina_

**Comment by GustavSt** [comment]
Any updates on this? Will this issue be solved with the next release?

**Comment by alvarotrigo** [comment]
@GustavSt nop.

**Comment by Akkuolina** [comment]
Peccato!
However, I have resolved with the solution that I have proposed (up)

Alvaro compliments!!! :)

**Comment by GustavSt** [comment]
I did do it that way. And in order for it to work for users resizing their browser window I destroyed fullpage and recreated it when `afterResponsive` is triggered. That event is not documented by the way!

**Comment by alvarotrigo** [comment]
> That event is not documented by the way!

True. Introduced in the previous release :)

**Comment by facundosegura** [comment]
Hello GustavSt
Soy novato en esto, Quisiera saber como se escribe el código:

// before running the fullpage
var scrolloverflowed;
if ($('#fullpage').width() < 1024) { // check the width (or window.width() or other method)
scrolloverflowed = false;
} else {
scrolloverflowed = true;
}

// fullpage init
$('#fullpage').fullpage({
...
...
scrollOverflow: scrolloverflowed,
...
...
});

muchas gracias!!

**Comment by Akkuolina** [comment]
> Hello GustavSt
> Soy novato en esto, Quisiera saber como se escribe el código:

Hi facundosegura,
I'm not GustavSt, You have mistaken person :)
Anyway ... if you follow the Alvaro's guide, you'll find the exact point in which to include the code that I had reported up.

Greetings...
A.

**Comment by facundosegura** [comment]
Hola Akkuolina
Soy novato en esto, Quisiera saber como se escribe el código completo:
// before running the fullpage
var scrolloverflowed;
if ($('#fullpage').width() < 1024) { // check the width (or window.width() or other method)
scrolloverflowed = false;
} else {
scrolloverflowed = true;
}

// fullpage init
$('#fullpage').fullpage({
...
...
scrollOverflow: scrolloverflowed,
...
...
});
Porque sí no funciona.
muchas gracias!!

**Comment by Akkuolina** [comment]
Hello facundosegura,

You can refer to any example of fullpage, for example:
https://github.com/alvarotrigo/fullPage.js/blob/master/examples/demoPage.html

At line 58, you can replace all the code in `$(document).ready` with this below:

```
$(document).ready(function() {
	var scrolloverflowed;
	if ($('#fullpage').width() < 1024) { // check the width (window.width() or other method)
		scrolloverflowed = false;
	} else {
		scrolloverflowed = true;
	}
	$('#fullpage').fullpage({
		sectionsColor: ['#1bbc9b', '#4BBFC3', '#7BAABE', 'whitesmoke', '#ccddff'],
		anchors: ['firstPage', 'secondPage', '3rdPage', '4thpage', 'lastPage'],
		menu: '#menu',
		scrollingSpeed: 1000,
		scrollOverflow: scrolloverflowed,
	});
});
```

Bye bye
Akkuolina

**Comment by facundosegura** [comment]
Gracias Akkuolina, pero no funciona en ios en ipad, cuando el texto es largo no scrola
alguna solución?

**Comment by meceware** [comment]
Hi,

The javascript workaround mentioned here actually did not help for me, since when the orientation changes, scroll overflow does not work even if it should work. (Like on tablets)

Instead of a javascript call, I added some CSS to disable overflow scroll when the page is responsive. I

/* Change 965 to the number you put as responsive */
@media (max-width:965px)
{
    /* Disable scroll overflow by changing height */
    .fp-auto-height-responsive > .fp-scrollable {
        height: auto !important;
    }
    /* Disable scrolling transform */
    .fp-auto-height-responsive > .fp-scrollable > .fp-scroller {
        transform: none !important;
    }
    /* Hide scrollbars */
    .wpex-vc-row-wrap.fp-auto-height-responsive > .fp-scrollable > .iScrollVerticalScrollbar{
        display: none !important;
    }
}

It works on my trials but I would like to get opinions if this would be a better solution to this problem.

**Comment by rostik** [comment]
@media only screen and (max-width:1023px) {
    .fp-scrollable {
    	height: auto !important;
    }
}

this fixed it for me

**Comment by alvarotrigo** [comment]
Feature added on dev. To be merged in 2.9.5.
`scrollOverflow` won't be applied on responsive to sections containing the class `fp-auto-height-responsive`.
